### PR TITLE
Fix watch_order_book bug on testnet spot 

### PR DESCRIPTION
**Pull request by idan-ben-ami**
Issue: #16780

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @idan-ben-ami,
Thanks for your contribution we will review it soon!

File: ts/src/pro/binance.ts [hunk 1]
```diff
@@ -293,8 +293,10 @@ export default class binance extends binanceRest {
         //         ]
         //     }
         //
-        const index = client.url.indexOf ('/stream');
-        const marketType = (index >= 0) ? 'spot' : 'contract';
+        // Note: not calling self.market(message['s'])['type'] because it's very slow (10x slower)
+        const isTestnetSpot = client.url.indexOf ('testnet') > 0;
+        const isSpot = client.url.indexOf ('/stream.binance.') > 0;
+        const marketType = (isTestnetSpot || isSpot) ? 'spot' : 'contract';
```
Comment (carlosmiei) [inline ts/src/pro/binance.ts:299]: not sure we can do this, what if you;re fetching a swap orderbook from the futures testnet?

File: ts/src/pro/binance.ts [hunk 2]
```diff
@@ -293,8 +293,10 @@ export default class binance extends binanceRest {
         //         ]
         //     }
         //
-        const index = client.url.indexOf ('/stream');
-        const marketType = (index >= 0) ? 'spot' : 'contract';
+        // Note: not calling self.market(message['s'])['type'] because it's very slow (10x slower)
```
Comment (carlosmiei) [inline ts/src/pro/binance.ts:296]: did you measure it in absolute terms?

**Comment by carlosmiei** [comment]
@idan-ben-ami Will fix it here https://github.com/ccxt/ccxt/pull/17616 in a simpler way, thanks anyway for bringing this up.

File: ts/src/pro/binance.ts (hunk 1 above)
Comment (idan-ben-ami) [reply ts/src/pro/binance.ts:299]: Hence the '.' in the end of the string.

**Comment by idan-ben-ami** [comment]
> @idan-ben-ami Will fix it here #17616 in a simpler way, thanks anyway for bringing this up.

Why is it simpler? and why did you close this PR?

**Comment by carlosmiei** [comment]
@idan-ben-ami Sorry my bad, I thought yours was not working with swap markets, will merge it soon

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**